### PR TITLE
[ZEPPELIN-4763]. Use isolated interpreter for cron note job

### PR DESCRIPTION
**Pull request by zjffdu**
### What is this PR for?

This PR is to use isolated interpreter for cron note job. Cron job is for production job, so we should use dedicated interpreter for it instead of using shared interpreter with other notes or users.  Currently interpreter binding mode is determined by note or users (scoped or isolated), This PR introduce `ExecutionContext` which include other factors (isCronMode) to determine how we create interpreters.

### What type of PR is it?
[Feature]

### Todos
* [ ] - Task

### What is the Jira issue?
* https://issues.apache.org/jira/browse/ZEPPELIN-4763

### How should this be tested?
* Unit test is updated, and manually tested

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

File: zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/ExecutionContext.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.zeppelin.interpreter;
+
+public class ExecutionContext {
+
+  private String user;
+  private String noteId;
+  private boolean inCronMode;
```
Comment (Reamer) [inline zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/ExecutionContext.java:24]: You should add final, because you have no setter.

Comment (zjffdu) [reply zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/ExecutionContext.java:24]: Good catch，will fix it.

File: zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterSetting.java [hunk 2]
```diff
@@ -422,16 +424,23 @@ public String getGroup() {
     return group;
   }
 
-  private String getInterpreterGroupId(String user, String noteId) {
+  private static DateTimeFormatter dtf = DateTimeFormatter.ofPattern("yyyy-MM-dd_HH-mm-ss");
```
Comment (Reamer) [inline zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterSetting.java:427]: Should also be an candidate for the `final` keyword.

**Comment by Reamer** [comment]
LGTM. Can you solve the merge conflict in `QuartzSchedulerService.java`.

**Pull request closed without merging**